pkg/util/util.go

```diff
@@ -1,12 +1,50 @@
 package util
 
 import (
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/pkg/errors"
 )
 
+// Mirrors path to a tmpfile if path points to a
+// file descriptor instead of actual file on filesystem
+// reason: operations with file descriptors are can lead
+// to edge cases where content on FD is not in a consumable
+// state after first consumption
+func MirrorToTempFileIfPathIsDescriptor(file string) string {
+	// one use-case is discussed here
+	// https://github.com/containers/buildah/issues/3070
+	if !strings.HasPrefix(file, "/dev/fd") {
+		return file
+	}
+	b, err := ioutil.ReadFile(file)
+	if err != nil {
+		// if anything goes wrong return original path
+		return file
+	}
+	tmpfile, err := ioutil.TempFile(os.TempDir(), "buildah-temp-file")
+	if err != nil {
+		return file
+	}
+	if _, err := tmpfile.Write(b); err != nil {
+		// if anything goes wrong return original path
+		return file
+	}
+
+	return tmpfile.Name()
+}
+
+// Removes any temp buildah file if it was created in the first place
+func RemoveTempFileByPath(path string) {
+	// check if path has prefix tmpDir+builah-temp-file
+	if strings.HasPrefix(path, filepath.Join(os.TempDir(), "buildah-temp-file")) {
```

Comment:
I don't like that logic (as expressed in the c/common PR). I'd find it cleaner if `MirrorToTempFileIfPathIsDescriptor` had a 2nd return value indicating whether the authfile should be removed after usage or not.

Reply:
fixed in recent commit.

Reply:
To explicit: Please remove this function and replace the caller with `os.Remove(path)`. The caller now knows whether to remove the file or not, so don't need the redundant prefix check. Less magic, less code :)

Reply:
@vrothberg ah my bad fixed it.